### PR TITLE
Include the Pounders algorithm in estimagic 

### LINKED ISSUE #15: Add pounders algorithm to our optimizers
https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Tao/TAOPOUNDERS.html

Best way to do that is probably [petsc4py](https://anaconda.org/conda-forge/petsc4py/files) but the [documentation](https://www.mcs.anl.gov/petsc/petsc4py-current/docs/usrman/overview.html#petsc-efficient) is very incomplete. 

The relevant import seems to be: `from petsc4py.PETSc import TAO`

### PR DESCRIPTION
**Pull request by mo2561057**
closes #15 

Add the pounders algorithm from the toolkit of advanced optimization. This will be the default optimizer for moment based estimation.

### CODE REVIEW

**Comment by janosg** [comment]
There is no windows version of petsc4py on conda-forge (or on any other channel). Quick poll on how we deal with this @hmgaudecker @peisenha @tobiasraabe @roecla 

- Try to add it to conda forge
- Just not support pounders on windows (probably by using [selectors](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors) in `meta.yaml` 
- Other ideas?

**Comment by hmgaudecker** [comment]
Is it only petsc4py? That is a thin wrapper, I believe, might be worth asking the maintainer (Lisandro Dalcin, he was very helpful when I had some issues since time back) of he sees any problems.

Now, if we are talking of petsc not being available, stop right here. That is a huge animal written in C or C++...

**Comment by janosg** [comment]
There is also no petsc for windows on conda. BTW: The first sentence on the official installation guide of petsc for windows is: `Are you sure you want to use Microsoft Windows?` :-D

**Comment by hmgaudecker** [comment]
Good one. 

Just point people here: https://docs.microsoft.com/en-us/windows/wsl/install-win10

**Comment by tobiasraabe** [comment]
I agree with Hans-Martin. People are usually on windows 10 so they should use Wsl. It is easy to install and run. And it only affects a special subset of estimagics usability. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by janosg** [comment]
We have to postpone merging until we find a way to make the conda selectors work. I posted under a related [Issue](https://github.com/conda/conda-build/issues/2093) and hope we get a quick reply. 

In the worst case we have to build the windows package on a remote windows computer.

File: environment.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ dependencies:
   - seaborn
   - jupyter
   - fuzzywuzzy
+  - mpi
+  - petsc4py
```
Comment (hmgaudecker) [inline environment.yml]: Make sure you require >= 3.11, I just got bitten by the silent SeparableObjective -> Residual rename elsewhere...

Comment (janosg) [reply environment.yml]: Will do. I'll finish this over the weekend.

**Comment by janosg** [comment]
I tried to solve our windows installation problems by uploading a fake `petsc4py` package to my anaconda channel (windows versions only). If this works on @tobiasraabe 's windows machine, we can merge. I know this is not ideal, but all the other solutions would be much more work.

**Pull request merged**